docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_geo_restriction.mdx

```diff
@@ -23,6 +23,323 @@ SOC2, GDPR
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all CloudFront distributions: Use the following AWS CLI command to list all the CloudFront distributions in your AWS account:
+
+   ```
+   aws cloudfront list-distributions --query 'DistributionList.Items[*].Id' --output text
+   ```
+   This command will return a list of IDs for all the CloudFront distributions.
+
+3. Get the details of each CloudFront distribution: For each ID returned in the previous step, run the following command to get the details of the distribution:
+
+   ```
+   aws cloudfront get-distribution --id <distribution-id>
+   ```
+   Replace `<distribution-id>` with the ID of the distribution you want to check. This command will return a JSON object with the details of the distribution.
+
+4. Check for Geo Restriction: In the JSON object returned in the previous step, look for the `Restrictions` field. If the `GeoRestriction` field under `Restrictions` is not present or if the `RestrictionType` is not set to `whitelist` or `blacklist`, then the CloudFront distribution does not have Geo Restriction enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install and configure Boto3 to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to allow Boto3 to interact with AWS services on your behalf. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID='your_access_key'
+   AWS_SECRET_ACCESS_KEY='your_secret_key'
+   ```
+
+2. Import the necessary modules and create a CloudFront client:
+   You need to import Boto3 and create a CloudFront client to interact with the CloudFront service.
+   ```python
+   import boto3
+
+   # Create CloudFront client
+   cloudfront = boto3.client('cloudfront')
+   ```
+
+3. List all CloudFront distributions and check for Geo Restriction:
+   You can list all CloudFront distributions using the `list_distributions` method and then check if Geo Restriction is enabled for each distribution.
+   ```python
+   # List all distributions
+   distributions = cloudfront.list_distributions()
+
+   for distribution in distributions['DistributionList']['Items']:
+       # Get the distribution config
+       distribution_config = cloudfront.get_distribution_config(Id=distribution['Id'])
+
+       # Check if Geo Restriction is enabled
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>